schema/xmlelement.cpp

```diff
@@ -104,4 +104,10 @@ Annotation::List XmlElement::annotations() const
 {
     return d->mAnnotations;
 }
+
+bool XmlElement::operator==(const XmlElement &other) const
+{
+    return (isNull() == other.isNull() && name() == other.name());
```

Comment:
Since `isNull()` is mostly a test on `name()`, it seems redundant to test it here.

```suggestion
    return name() == other.name();
```